packages/dev/core/src/Rendering/boundingBoxRenderer.ts

```diff
@@ -162,6 +178,29 @@ export class BoundingBoxRenderer implements ISceneComponent {
     private _uniformBufferFront: UniformBuffer;
     private _uniformBufferBack: UniformBuffer;
     private _renderPassIdForOcclusionQuery: number;
+    /**
+     * Internal buffer for instanced rendering
+     * @internal
+     */
+    private _matrixBuffer: Nullable<Buffer> = null;
+    private _matrices: Nullable<Float32Array> = null;
+
+    /**
+     * Internal state of whether instanced rendering enabled
+     * @internal
+     */
+    protected _useInstances = false;
+    /**
+     * Choose whether dynamic vertex buffer should be used for instanced rendering.
+     * If true, the buffer would be updated instead of recreated when rendering a frame,
+     * this could reduce memory pressure.
+     */
+    public updatable = true;

```

Comment:
I don't think we need this variable? Just keep the "updatable=true" case in the code, as I can't see why a user would want to set `updatable=false` (recreating the vertex buffer every frame will likely always be slower than updating it).

Reply:
The docs of [Mesh](https://doc.babylonjs.com/typedoc/classes/BABYLON.Mesh#thinInstanceSetBuffer) said that using static buffer could have better performances, but in recent tests on a chrome, recreating it leads to more memory pressure, so this is kept and defaults to `true`

Reply:
Yes, performance is better when `static=true` if you create the buffers once and reuse them. But in your case, you recreate the buffers every frame when `updatable=false`. So even if the buffers are created with `static=false`, it's probably slower than creating the buffers once with `static=false` and updating them afterwards. You should run some tests to make sure, as we don't want to keep a code path that will never be used.

Reply:
OK, I'll run some profiling on chrome and firefox, but I does not have access to safari profiler. Also, since the buffer would be recreated when size change or `useInstances` change, the buffer recreation code is needed, so there is few extra code for this option.

Reply:
Summary:
| Configuration         | Browser  | _renderInstanced | Heap     |
|-----------------------|----------|-------------------|----------|
| `updatable=true`      | Chrome   | 65.1%             | 102 MB   |
|                       | Firefox  | 69%               | 156.33 MB|
| `updatable=false`     | Chrome   | 69.0%             | 103 MB   |
|                       | Firefox  | 72%               | 154.20 MB|

<details><summary>Details</summary>
<p>

[`updatable`=`true`](https://playground.babylonjs.com/?snapshot=refs%2Fpull%2F15911%2Fmerge#NRNVQA):
Chrome:
_renderInstanced: 65.1%
Heap: 102MB
![performance](https://github.com/user-attachments/assets/18d1f003-bfde-41c2-9d04-116033fc13f3)
![buttom-up](https://github.com/user-attachments/assets/b53d647e-3849-45b3-9d09-3ecb5386548a)
![heap](https://github.com/user-attachments/assets/a857ea26-fb23-4694-a737-ec03fda89769)

Firefox:
_renderInstanced: 69%
Heap: 156.33MB
![call tree](https://github.com/user-attachments/assets/fb12f423-38ef-4689-8ac0-6bb1cc4d5458)
![flame graph](https://github.com/user-attachments/assets/a810c692-651a-4478-9acb-53469f8a8c2a)
![heap](https://github.com/user-attachments/assets/138405f9-220d-404b-905a-83385169972c)

[`updatable`=`false`](https://playground.babylonjs.com/?snapshot=refs%2Fpull%2F15911%2Fmerge#NRNVQA#1):
Chrome:
_renderInstanced: 69.0%
Heap: 103MB
![performance](https://github.com/user-attachments/assets/64a826b9-73b5-4f0a-87a8-5e85d8702a3a)
![buttom-up](https://github.com/user-attachments/assets/489f43ec-0d9d-4156-a375-dc9af0a2124a)
![heap](https://github.com/user-attachments/assets/10b135e4-38fd-4c3a-92b3-5e49f4b5e2a2)

Firefox:
_renderInstanced: 72%
Heap: 154.20MB
![call tree](https://github.com/user-attachments/assets/256e1a57-412d-4e09-b086-6dfbebe784ce)
![flame graph](https://github.com/user-attachments/assets/c095f2d6-5dfb-4245-9366-5099e1f2e951)
![heap](https://github.com/user-attachments/assets/5380ede9-b8d1-4088-882c-e04e60d3a7a2)



</p>
</details>